core/src/apps/solana/transaction/instruction.py

```diff
@@ -50,9 +50,23 @@ def parse_instruction_data(
         parsed_data = {}
         for property_template in property_templates:
             is_included = True
+            # Property cannot be both optional and nullable as if there are multiple such properties in 1 instruction
+            # it could lead to non-deterministic parsing
+            # it is validated by mako template
+            assert not (property_template.is_nullable and property_template.is_optional)
+
             if property_template.is_optional:
                 is_included = True if reader.get() == 1 else False
 
+            if property_template.is_nullable:
+                is_included = True if reader.get() == 1 else False
+                if not is_included:
+                    # A default (null) value is included in the parsed message for nullable properties
+                    # (e.g. new_authority in Set Authority instruction) even if the is_included flag is not set.
+                    # We can safely discard this value.
+                    if reader.remaining_count() != 0:
+                        property_template.parse(reader)
```

Comment:
if we can safely discard, we should not be checking for `remaining_count()` here, right? consider a value that is "included" but then the data ends. we _want_ the eof fail in that case